src/main/java/org/ow2/proactive/catalog/repository/BucketGrantRepository.java

```diff
@@ -41,16 +41,32 @@
 
     List<BucketGrantEntity> deleteAllByBucketEntityId(long bucketId);
 
-    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1 AND bge.grantee = ?2 AND bge.granteeType='user'")
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1 AND bge.grantee = ?2 AND bge.granteeType='user' AND bge.accessType<>'noAccess'")
     BucketGrantEntity findBucketGrantByUsername(long bucketId, String username);
 
-    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1 AND bge.grantee = ?2 AND bge.granteeType='group'")
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1 AND bge.grantee = ?2 AND bge.granteeType='user'")
+    BucketGrantEntity findBucketGrantByUsernameForUpdate(long bucketId, String username);
```

Comment:
For the func name `findBucketGrantByUsername` and `findBucketGrantByUsernameForUpdate`, I thought `findAccessibleBucketGrantByUsername` and `findAllBucketGrantByUsername` might be a bit more clear.

Reply:
Ok Thanks, I modified them.